README.md

```diff
@@ -1,51 +1,92 @@
-# nginx_otel
+# NGINX Native OpenTelemetry (OTel) Module
 
-This project provides support for OpenTelemetry distributed tracing in Nginx, offering:
+## What is OpenTelemetry
+OpenTelemetry (OTel) is an observability framework for monitoring, tracing, troubleshooting, and optimizing applications. OTel enables the collection of telemetry data from a deployed application stack.
 
-- Lightweight and high-performance incoming HTTP request tracing
-- [W3C trace context](https://www.w3.org/TR/trace-context/) propagation
-- OTLP/gRPC trace export
-- Fully Dynamic Variable-Based Sampling
+## What is the Native NGINX Otel Module
+The `ngx_otel_module` dynamic module enables NGINX Open-Source or NGINX Plus to send telemetry data to an OTel collector. It provides support for [W3C trace context](https://www.w3.org/TR/trace-context/) propagation, OTLP/gRPC trace exports and offers several benefits over exiting OTel modules, including:
+
+### Better Performance ###
+3rd-party OTel implementations reduce performance of request processing by as much as 50% when tracing is enabled. The NGINX Native module limits this impact to approximately 10-15%.
+
+### Easy Provisioning ###
+Setup and configuration can be done right in NGINX configuration files.
+
+### Fully Dynamic Variable-Based Sampling ###
+The module provides the ability to trace a particular session by cookie/token. [NGINX Plus](https://www.nginx.com/products/nginx/), available as part of a [commercial subscription](https://www.nginx.com/products/), enables dynamic module control via the [NGINX Plus API](http://nginx.org/en/docs/http/ngx_http_api_module.html) and [key-value store](http://nginx.org/en/docs/http/ngx_http_keyval_module.html) modules.
 
 ## Building
+Follow these steps to build the `ngx_otel_module` dynamic module on Ubuntu or Debian based systems:
 
-Install build tools and dependencies:
+Install build tools and dependencies.
 ```bash
-  $ sudo apt install cmake build-essential libssl-dev zlib1g-dev libpcre3-dev
-  $ sudo apt install pkg-config libc-ares-dev libre2-dev # for gRPC
+sudo apt install cmake build-essential libssl-dev zlib1g-dev libpcre3-dev
+sudo apt install pkg-config libc-ares-dev libre2-dev # for gRPC
 ```
 
-Configure Nginx:
+Clone the NGINX repository.
 ```bash
-  $ ./configure --with-compat
+git clone https://github.com/nginx/nginx.git
 ```
 
-Configure and build Nginx OTel module:
+Configure NGINX to create objects necessary for dynamic module compilation. Objects will be placed into the `nginx/objs` directory.
```

Comment:
This step won't place any objects into 'objs' directory, nor will subsequent step from this guide. We actually need a header file generated during this step, but I'm not sure we have to elaborate on that here.

Reply:
Ok, makes sense. I agree that we don't need to get that specific, so I changed to "files". https://github.com/nginxinc/nginx-otel/pull/7/commits/ecd17610216867c0d889ba7df69d7d40dc489830